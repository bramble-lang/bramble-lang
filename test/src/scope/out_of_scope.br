fn my_main() {
    let x: i64 := if (true) {
        let y: i64 := 5;
        y
    } else {
        y
    };
    return;
}