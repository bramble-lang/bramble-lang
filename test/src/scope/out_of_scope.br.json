[{"id": 1, "stage": "lexer", "source": [0, 2], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [3, 10], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [10, 11], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [13, 15], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [16, 19], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [20, 21], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [26, 29], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [30, 31], "ok": "Identifier"},
{"id": 10, "stage": "lexer", "source": [31, 32], "ok": "Operator"},
{"id": 11, "stage": "lexer", "source": [33, 36], "ok": "Primitive"},
{"id": 12, "stage": "lexer", "source": [37, 39], "ok": "Operator"},
{"id": 13, "stage": "lexer", "source": [40, 42], "ok": "Keyword"},
{"id": 14, "stage": "lexer", "source": [43, 44], "ok": "Operator"},
{"id": 15, "stage": "lexer", "source": [44, 48], "ok": "Boolean"},
{"id": 16, "stage": "lexer", "source": [48, 49], "ok": "Operator"},
{"id": 17, "stage": "lexer", "source": [50, 51], "ok": "Operator"},
{"id": 18, "stage": "lexer", "source": [60, 63], "ok": "Keyword"},
{"id": 19, "stage": "lexer", "source": [64, 65], "ok": "Identifier"},
{"id": 20, "stage": "lexer", "source": [65, 66], "ok": "Operator"},
{"id": 21, "stage": "lexer", "source": [67, 70], "ok": "Primitive"},
{"id": 22, "stage": "lexer", "source": [71, 73], "ok": "Operator"},
{"id": 23, "stage": "lexer", "source": [74, 75], "ok": "Number"},
{"id": 24, "stage": "lexer", "source": [75, 76], "ok": "Operator"},
{"id": 25, "stage": "lexer", "source": [85, 86], "ok": "Identifier"},
{"id": 26, "stage": "lexer", "source": [91, 92], "ok": "Operator"},
{"id": 27, "stage": "lexer", "source": [93, 97], "ok": "Keyword"},
{"id": 28, "stage": "lexer", "source": [98, 99], "ok": "Operator"},
{"id": 29, "stage": "lexer", "source": [108, 109], "ok": "Identifier"},
{"id": 30, "stage": "lexer", "source": [114, 115], "ok": "Operator"},
{"id": 31, "stage": "lexer", "source": [115, 116], "ok": "Operator"},
{"id": 32, "stage": "lexer", "source": [121, 127], "ok": "Keyword"},
{"id": 33, "stage": "lexer", "source": [128, 129], "ok": "Number"},
{"id": 34, "stage": "lexer", "source": [129, 130], "ok": "Operator"},
{"id": 35, "stage": "lexer", "source": [131, 132], "ok": "Operator"},
{"id": 41, "parent_id": 39, "stage": "parser", "source": [16, 19], "ok": "Primitive Type"},
{"id": 39, "parent_id": 38, "stage": "parser", "source": [0, 19], "ok": "Routine Declaration"},
{"id": 45, "parent_id": 44, "stage": "parser", "source": [33, 36], "ok": "Primitive Type"},
{"id": 44, "parent_id": 43, "stage": "parser", "source": [30, 36], "ok": "Identifier Declaration"},
{"id": 66, "parent_id": 61, "stage": "parser", "source": [44, 48], "ok": "Boolean"},
{"id": 61, "parent_id": 59, "stage": "parser", "source": [44, 48]},
{"id": 59, "parent_id": 58, "stage": "parser", "source": [44, 48]},
{"id": 58, "parent_id": 57, "stage": "parser", "source": [44, 48]},
{"id": 57, "parent_id": 56, "stage": "parser", "source": [44, 48]},
{"id": 56, "parent_id": 55, "stage": "parser", "source": [44, 48]},
{"id": 55, "parent_id": 54, "stage": "parser", "source": [44, 48]},
{"id": 71, "parent_id": 70, "stage": "parser", "source": [67, 70], "ok": "Primitive Type"},
{"id": 70, "parent_id": 69, "stage": "parser", "source": [64, 70], "ok": "Identifier Declaration"},
{"id": 83, "parent_id": 79, "stage": "parser", "source": [74, 75], "ok": "Number"},
{"id": 79, "parent_id": 77, "stage": "parser", "source": [74, 75]},
{"id": 77, "parent_id": 76, "stage": "parser", "source": [74, 75]},
{"id": 76, "parent_id": 75, "stage": "parser", "source": [74, 75]},
{"id": 75, "parent_id": 74, "stage": "parser", "source": [74, 75]},
{"id": 74, "parent_id": 73, "stage": "parser", "source": [74, 75]},
{"id": 73, "parent_id": 69, "stage": "parser", "source": [74, 75]},
{"id": 69, "parent_id": 68, "stage": "parser", "source": [60, 75], "ok": "Let Binding"},
{"id": 68, "parent_id": 67, "stage": "parser", "source": [60, 76], "ok": "Statement Bind"},
{"id": 96, "parent_id": 95, "stage": "parser", "source": [85, 86], "ok": "Path"},
{"id": 95, "parent_id": 93, "stage": "parser", "source": [85, 86], "ok": "Identifier"},
{"id": 93, "parent_id": 91, "stage": "parser", "source": [85, 86]},
{"id": 91, "parent_id": 90, "stage": "parser", "source": [85, 86]},
{"id": 90, "parent_id": 89, "stage": "parser", "source": [85, 86]},
{"id": 89, "parent_id": 88, "stage": "parser", "source": [85, 86]},
{"id": 88, "parent_id": 87, "stage": "parser", "source": [85, 86]},
{"id": 87, "parent_id": 84, "stage": "parser", "source": [85, 86]},
{"id": 107, "parent_id": 106, "stage": "parser", "source": [85, 86], "ok": "Path"},
{"id": 106, "parent_id": 104, "stage": "parser", "source": [85, 86], "ok": "Identifier"},
{"id": 104, "parent_id": 102, "stage": "parser", "source": [85, 86]},
{"id": 102, "parent_id": 101, "stage": "parser", "source": [85, 86]},
{"id": 101, "parent_id": 100, "stage": "parser", "source": [85, 86]},
{"id": 100, "parent_id": 99, "stage": "parser", "source": [85, 86]},
{"id": 99, "parent_id": 98, "stage": "parser", "source": [85, 86]},
{"id": 98, "parent_id": 67, "stage": "parser", "source": [85, 86]},
{"id": 67, "parent_id": 54, "stage": "parser", "source": [50, 92], "ok": "Expression Block"},
{"id": 122, "parent_id": 121, "stage": "parser", "source": [108, 109], "ok": "Path"},
{"id": 121, "parent_id": 119, "stage": "parser", "source": [108, 109], "ok": "Identifier"},
{"id": 119, "parent_id": 117, "stage": "parser", "source": [108, 109]},
{"id": 117, "parent_id": 116, "stage": "parser", "source": [108, 109]},
{"id": 116, "parent_id": 115, "stage": "parser", "source": [108, 109]},
{"id": 115, "parent_id": 114, "stage": "parser", "source": [108, 109]},
{"id": 114, "parent_id": 113, "stage": "parser", "source": [108, 109]},
{"id": 113, "parent_id": 110, "stage": "parser", "source": [108, 109]},
{"id": 133, "parent_id": 132, "stage": "parser", "source": [108, 109], "ok": "Path"},
{"id": 132, "parent_id": 130, "stage": "parser", "source": [108, 109], "ok": "Identifier"},
{"id": 130, "parent_id": 128, "stage": "parser", "source": [108, 109]},
{"id": 128, "parent_id": 127, "stage": "parser", "source": [108, 109]},
{"id": 127, "parent_id": 126, "stage": "parser", "source": [108, 109]},
{"id": 126, "parent_id": 125, "stage": "parser", "source": [108, 109]},
{"id": 125, "parent_id": 124, "stage": "parser", "source": [108, 109]},
{"id": 124, "parent_id": 109, "stage": "parser", "source": [108, 109]},
{"id": 109, "parent_id": 54, "stage": "parser", "source": [98, 115], "ok": "Expression Block"},
{"id": 54, "parent_id": 53, "stage": "parser", "source": [40, 115], "ok": "If Expression"},
{"id": 53, "parent_id": 51, "stage": "parser", "source": [40, 115]},
{"id": 51, "parent_id": 50, "stage": "parser", "source": [40, 115]},
{"id": 50, "parent_id": 49, "stage": "parser", "source": [40, 115]},
{"id": 49, "parent_id": 48, "stage": "parser", "source": [40, 115]},
{"id": 48, "parent_id": 47, "stage": "parser", "source": [40, 115]},
{"id": 47, "parent_id": 43, "stage": "parser", "source": [40, 115]},
{"id": 43, "parent_id": 42, "stage": "parser", "source": [26, 115], "ok": "Let Binding"},
{"id": 42, "parent_id": 38, "stage": "parser", "source": [26, 116], "ok": "Statement Bind"},
{"id": 162, "parent_id": 158, "stage": "parser", "source": [128, 129], "ok": "Number"},
{"id": 158, "parent_id": 156, "stage": "parser", "source": [128, 129]},
{"id": 156, "parent_id": 155, "stage": "parser", "source": [128, 129]},
{"id": 155, "parent_id": 154, "stage": "parser", "source": [128, 129]},
{"id": 154, "parent_id": 153, "stage": "parser", "source": [128, 129]},
{"id": 153, "parent_id": 152, "stage": "parser", "source": [128, 129]},
{"id": 152, "parent_id": 151, "stage": "parser", "source": [128, 129]},
{"id": 151, "parent_id": 38, "stage": "parser", "source": [121, 129], "ok": "Return"},
{"id": 38, "parent_id": 36, "stage": "parser", "source": [0, 132], "ok": "Funtion Definition"},
{"id": 36, "stage": "parser", "source": [0, 132], "ok": "File Module"},
{"id": 165, "stage": "canonize-item-path", "source": [0, 132], "ok": "$out_of_scope::out_of_scope"},
{"id": 166, "stage": "canonize-item-path", "source": [0, 132], "ok": "$out_of_scope::out_of_scope::out_of_scope"},
{"id": 167, "stage": "canonize-item-path", "source": [0, 132], "ok": "$out_of_scope::out_of_scope::my_main"},
{"id": 171, "parent_id": 170, "stage": "type-resolver", "source": [44, 48], "ok": "bool"},
{"id": 174, "parent_id": 173, "stage": "type-resolver", "source": [74, 75], "ok": "i64"},
{"id": 173, "parent_id": 172, "stage": "type-resolver", "source": [60, 76], "ok": "i64"},
{"id": 175, "parent_id": 172, "stage": "type-resolver", "source": [85, 86], "ok": "i64", "ref": [60, 76]},
{"id": 172, "parent_id": 170, "stage": "type-resolver", "source": [50, 92], "ok": "i64"},
{"id": 177, "parent_id": 176, "stage": "type-resolver", "source": [108, 109], "error": "L6: Could not find definition for y in this scope"},
{"id": 176, "parent_id": 170, "stage": "type-resolver", "source": [108, 109], "error": "L6: Could not find definition for y in this scope"},
{"id": 170, "parent_id": 169, "stage": "type-resolver", "source": [108, 109], "error": "L6: Could not find definition for y in this scope"},
{"id": 169, "parent_id": 168, "stage": "type-resolver", "source": [108, 109], "error": "L6: Could not find definition for y in this scope"},
{"id": 168, "stage": "type-resolver", "source": [108, 109], "error": "L6: Could not find definition for y in this scope"}]