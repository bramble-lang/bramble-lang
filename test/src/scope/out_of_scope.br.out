Error: L6: Could not find definition for y in this scope
